### PR TITLE
libsoup: support cross by disabling gobject-introspection and vala

### PR DESCRIPTION
**Pull request by Mindavi**
Unfortunately this is a mass rebuild since the introspection flag is added :(.

###### Motivation for this change

Make more cross compilation possible. This is one of multiple patches to disable introspection. We lose some features when doing that, but at least we can make cross-compilation possible for the packages that have introspection disabled.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I only built this on master, not (yet) on staging. Should now build both as native and at least for aarch64-multiplatform. I will start a build for both (at least for cross, the other shouldn't be different) on staging, but that'll take a while.

I've also applied nixpkgs-fmt on this.

### CODE REVIEW

File: pkgs/development/libraries/libsoup/default.nix [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+, meson
+, ninja
+, pkg-config
+, gnome
+, libsysprof-capture
+, gnomeSupport ? true
+, sqlite
+, glib-networking
+, gobject-introspection
+, withIntrospection ? stdenv.buildPlatform == stdenv.targetPlatform
```
Comment (Mindavi) [inline pkgs/development/libraries/libsoup/default.nix:15]: Oops, should be buildPlatform and hostPlatform, I think. Will fix that later.

**Review by 7c6f434c — APPROVED** [review]

File: pkgs/development/libraries/libsoup/default.nix (hunk 1 above)
Comment (Mindavi) [reply pkgs/development/libraries/libsoup/default.nix]: Fixed.

**Comment by Mindavi** [comment]
Converted to draft since it's better to not make the `withIntrospection` an input parameter. Rather, the check could be inlined or expressed in a let-expression. Will look at that later.

**Comment by 7c6f434c** [comment]
Inline is clearly worse. (I think `let` is also worse)

**Comment by Mindavi** [comment]
Let's keep it like this then. I don't have a strong opinion on this.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**